client/my-sites/customize/main.jsx

```diff
@@ -65,11 +66,12 @@ var Customize = React.createClass( {
 	},
 
 	componentWillReceiveProps: function( nextProps ) {
-		this.redirectIfNeeded( nextProps.menusUrl, nextProps.site );
+		this.redirectIfNeeded( nextProps.pathname );
 	},
 
-	redirectIfNeeded: function( menusUrl, site ) {
-		if ( site && menusUrl !== '/customize/menus/' + site.slug ) {
+	redirectIfNeeded: function( pathname ) {
+		const { menusUrl } = this.props;
+		if ( startsWith( pathname, '/customize/menus' ) && pathname !== menusUrl ) {
 			page( menusUrl );
```

Comment:
Oh, we might actually want to use [`page.redirect`](https://github.com/visionmedia/page.js#pageredirectpath) here. That leaves a cleaner Browser history...